lib/services/provided/autocomplete.js

```diff
@@ -39,29 +41,41 @@ function parseCompletions(results: CompleteReply, prefix: string) {
 
   if (metadata && metadata._jupyter_types_experimental) {
     const comps = metadata._jupyter_types_experimental;
-    if (
-      comps.length > 0 &&
-      comps[0].text != null &&
-      comps[0].start != null &&
-      comps[0].end != null
-    ) {
-      return _.map(comps, match => ({
-        text: match.text,
-        replacementPrefix: prefix.slice(match.start, match.end),
-        type: match.type,
-        iconHTML:
-          !match.type || match.type === "<unknown>" ? iconHTML : undefined
-      }));
+    if (comps.length > 0 && comps[0].text) {
```

Comment:
I don't understand why we are adding cursor logic. Can you explain a case where previous code was failing and you code is necessary?

Reply:
This is simply because IJulia offers completions with "experimental type info" but they are not bundled with `start` or `end` property.
Like [this commit message](https://github.com/nteract/hydrogen/pull/1727/commits/400686db0821e2eb210552d048850adf792cdd3a) says.

Reply:
Oh I see what this means now thanks 👍

Reply:
if it is missing then can we just set cursor_start to match.start to avoid duplicate splicing logic?

Reply:
Not sure why IPython offers `start` and `end`, but as far as they propose them I just respect them and use our `cursor_*` as fallback.